src/Stratis.Bitcoin.Features.RPC/RPCMiddleware.cs

```diff
@@ -39,10 +45,36 @@ public async Task InvokeAsync(HttpContext httpContext)
                 httpContext.Response.StatusCode = StatusCodes.Status401Unauthorized;
                 return;
             }
-            Exception ex = null;
+
+            JToken token = null;
             try
             {
-                await this.next.Invoke(httpContext);
+                var memoryStream = new MemoryStream();
+                await httpContext.Request.Body.CopyToAsync(memoryStream);
+                httpContext.Request.Body = memoryStream;
+                memoryStream.Position = 0;
+
+                token = JToken.Load(new JsonTextReader(new StreamReader(memoryStream)));
+                httpContext.Request.Body.Position = 0;
+            }
+            catch (JsonException)
+            {
+                token = null;
```

Comment:
Yes this code is a bit weird, but it is to avoid having to rewrite a bunch of unit tests that assume that this is the way it works. This code makes the code behave the same as it did previously if the json cannot be parsed. I should maybe add a github issue for this, really I think JsonExceptions should show up as RPC Parse Errors, which they currently don't do but that will require some heavy unit test refactoring.

Reply:
This is an example test that would need to be refactored (there are several similar in RPCMiddlewareTest.cs)
```
        [Fact]
        public void Invoke500WritesInternalErrorAndLogsResult()
        {
            this.response.StatusCode = StatusCodes.Status500InternalServerError;
            this.SetupValidAuthorization();
            this.InitializeFeatureContext();

            this.middleware.InvokeAsync(this.httpContext).Wait();

            this.httpContext.Response.Body.Position = 0;
            using (var reader = new StreamReader(this.httpContext.Response.Body))
            {
                string expected = string.Format("{{{0}  \"result\": null,{0}  \"error\": {{{0}    \"code\": -32603,{0}    \"message\": \"Internal error\"{0}  }}{0}}}", Environment.NewLine);
                Assert.Equal(expected, reader.ReadToEnd());
                Assert.Equal(StatusCodes.Status500InternalServerError, this.httpContext.Response.StatusCode);
                this.AssertLog(this.Logger, LogLevel.Error, "Internal error while calling RPC Method");
            }
        }
```